# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/Object.h
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/public/Object.h
--- a/js/public/Object.h	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/public/Object.h	Sat Mar 06 01:05:21 2021 +0000
@@ -56,7 +56,7 @@
  * compartment of this realm.
  */
 static MOZ_ALWAYS_INLINE Compartment* GetCompartment(JSObject* obj) {
-  Realm* realm = reinterpret_cast<shadow::Object*>(obj)->group->realm;
+  Realm* realm = reinterpret_cast<shadow::Object*>(obj)->shape->base->realm;
   return GetCompartmentForRealm(realm);
 }
 